builtin/arrayview.mbt

```diff
@@ -173,3 +173,92 @@ pub fn rev_foldi[A, B](
     acc
   }
 }
+
+///|
+/// Maps a function over the elements of the array view.
+///
+/// # Example
+/// ```
+/// let v = [3, 4, 5]
+/// let v2 = v[1:].map(fn (x) {x + 1})
+/// assert_eq!(v2, [5, 6])
+/// ```
+pub fn map[T, U](self : ArrayView[T], f : (T) -> U) -> Array[U] {
+  if self.length() == 0 {
+    return []
+  }
+  let arr = Array::make_uninit(self.length())
+  for i, v in self {
+    arr.unsafe_set(i, f(v))
+  }
+  arr
+}
+
+///|
+/// Maps a function over the elements of the array view in place.
+///
+/// # Example
+/// ```
+/// let v = [3, 4, 5]
+/// v[1:].map_inplace(fn (x) {x + 1})
+/// assert_eq!(v, [3, 5, 6])
+/// ```
+pub fn map_inplace[T](self : ArrayView[T], f : (T) -> T) -> Unit {
+  for i, v in self {
+    self[i] = f(v)
+  }
+}
+
+///|
+/// Maps a function over the elements of the array view with index.
+///
+/// # Example
+/// ```
+/// let v = [3, 4, 5]
+/// let v2 = v[1:].mapi(fn (i, x) {x + i})
+/// assert_eq!(v2, [3, 4, 6])
```

Comment:
the CI failed because the test doc is incorrect

Reply:
Sorry about that, @bobzhang !  
I forgot about `moon test --doc`.
It should be fixed now.